pylzy/lzy/api/v1/remote/workflow_service_client.py

```diff
@@ -138,9 +148,29 @@ async def start_workflow(self, name: str, storage: Storage, storage_name: str) -
         else:
             raise ValueError(f"Invalid storage credentials type {type(storage.credentials)}")
 
-        res: StartWorkflowResponse = await self.__stub.StartWorkflow(
-            StartWorkflowRequest(workflowName=name, snapshotStorage=s, storageName=storage_name)
-        )
+        try:
+            res: StartWorkflowResponse = await self.__stub.StartWorkflow(
+                StartWorkflowRequest(workflowName=name, snapshotStorage=s, storageName=storage_name)
+            )
+        except AioRpcError as e:
+            data = e.trailing_metadata().get_all("x-supported-client-versions")
+            if data and len(data) > 0:
+                try:
+                    supported_versions = json.loads(data[0])
+                except Exception as ex:
+                    _LOG.warning("Cannot parse supported versions from server metadata: %s", ex)
+                    raise e
+
+                minimal_version = supported_versions.get("minimal_supported_version")
+
+                if minimal_version is None:
+                    raise
+
+                raise BadClientVersion(
+                    f"This version of pylzy is unsupported."
+                    f" Please run 'pip install pylzy=={minimal_version}' to update pylzy")
```

Comment:
I think `>=` would be better.
Also, I don't see a clue about banned versions.

Reply:
> but I don't know how yet

AFAIK it must be something like this:

`pip install -U 'pylzy>={min},!={banned},!={banned}'`